### ISSUE TITLE
for loop breaks S1128

### ISSUE THREAD

**Issue by askazakov** [SonarSource/sonar-dotnet#4853]
### Description

for loop breaks S1128 (_Remove unnecessary 'using' https://rules.sonarsource.com/csharp/RSPEC-1128_)

### Repro steps
0. `mkdir demo && cd demo`
1. `dotnet new console --framework netcoreapp3.1`
2. `dotnet add demo.csproj package SonarAnalyzer.CSharp`
3.  `echo '[*.cs]\ndotnet_diagnostic.S1128.severity = error' > .editorconfig`
4.  replace Program.cs with
```csharp
using System;

namespace demo
{
    class Program
    {
        static void Main(string[] args)
        {
        }
    }
}
```
5. `dotnet build`
6. add 
```csharp
for (var index = 0; index < new string[] { }.Length; index++)
{
}
```
to Main

7. `dotnet build`

### Expected behavior

Both builds fail with `error S1128: Remove this unnecessary 'using'. `

### Actual behavior

step 7 compiles successfully

### Related information

dotnet --info
.NET SDK (reflecting any global.json):
 Version:   5.0.302
 Commit:    c005824e35

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  11.0
 OS Platform: Darwin
 RID:         osx.11.0-x64
 Base Path:   /usr/local/share/dotnet/sdk/5.0.302/

Host (useful for support):
  Version: 5.0.8
  Commit:  35964c9215

**Comment by csaba-sagi-sonarsource** [comment]
Hello @askazakov , thank you for reporting this issue. I can confirm it as a False Negative.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1891](https://sonarsource.atlassian.net/browse/NET-1891)

[NET-1891]: https://sonarsource.atlassian.net/browse/NET-1891?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
